prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -305,6 +313,20 @@ class Hl7Serializer(val metadata: Metadata) {
                 setComponent(terser, element, element.hl7Field, value)
             }
         }
+        // make sure all fields we're suppressing are empty
+        suppressedFields.forEach {
+            val pathSpec = formPathSpec(it)
+            terser.set(pathSpec, "")
+        }
+        // check for reporting facility overrides
+        if (!hl7Config?.reportingFacilityName.isNullOrEmpty()) {
+            val pathSpec = formPathSpec("MSH-4-1")
+            terser.set(pathSpec, hl7Config?.reportingFacilityName)
+        }
+        if (!hl7Config?.reportingFacilityId.isNullOrEmpty()) {
+            val pathSpec = formPathSpec("MSH-4-2")
+            terser.set(pathSpec, hl7Config?.reportingFacilityId)
+        }
     }
```

Comment:
At the end of writing the message, we want to take one more pass through and set the suppressed HL7 fields to blank, and then force the aggregator fields to be what is passed in.